File: searching/BinarySearch/BinarySearchList.py
import math 

def binarySearch(haystack, needle):
    lo = 0
    hi = len(haystack)
    
    while lo < hi:
        mid = (math.floor(lo + (hi-lo) / 2))
        val = haystack[mid]
        
        if (val == needle):
            return True
        elif (val > needle):
            hi = mid
        else:
            lo = mid + 1
    return False

File: searching/BinarySearch/BinarySearchList.go
package main

func BinarySearchList(
	haystack []int,
	needle int) bool {
	lo := 0
	hi := len(haystack)

	for lo < hi {
		mid := lo + (hi-lo)/2
		val := haystack[mid]

		if val == needle {
			return true
		} else if val > needle {
			hi = mid
		} else {
			lo = mid + 1
		}
	}
	return false
}
